go/cmd/dolt/commands/branch.go

```diff
@@ -311,18 +312,51 @@ func createBranch(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgPars
 		return 1
 	}
 
+	var remote string
+	var remoteBranch string
 	newBranch := apr.Arg(0)
 	startPt := "head"
-
 	if apr.NArg() == 2 {
 		startPt = apr.Arg(1)
 	}
 
+	trackVal, setTrackUpstream := apr.GetValue(cli.TrackFlag)
+	if setTrackUpstream {
+		if trackVal == "inherit" {
+			return HandleVErrAndExitCode(errhand.BuildDError("--track='inherit' is not supported yet").Build(), usage)
+		} else if trackVal == "direct" && apr.NArg() != 2 {
+			return HandleVErrAndExitCode(errhand.BuildDError("invalid arguments").Build(), usage)
+		}
+		if apr.NArg() == 2 {
+			newBranch = apr.Arg(0)
+			startPt = apr.Arg(1)
+		} else {
+			newBranch = trackVal
```

Comment:
It seems like if there's only one argument, that has to be the `newBranch` name, since that is a required arg, and startPt would be the current working branch `head` by default? That seems to be how Git handles it in a quick test. 

but... I know you mentioned we don't support a branch tracking a local branch, so maybe we require two args – a new branch name and a starting point that is a remote branch ref?

Reply:
Yes, if `--track` option is used without defining either `direct` or `inherit`, the flag value will take the new branch name value since it expects an input value. Not super sure if this is the correct way to handle optional input for `--track`. I did similar approach for checkout.

Reply:
ah... thank you. This makes sense now! I totally missed that on Friday afternoon. Might be worth a short comment in there, just in case other people have a hard time reading that "newBranch = trackVal" and thinking it still contains the track val.

Reply:
I'll add comment to clarify this :)